File: backend/src/models/cluster.rs
use serde::{Deserialize, Serialize};
use sqlx::FromRow;
use uuid::Uuid;
use chrono::{DateTime, Utc};

#[derive(Debug, Serialize, Deserialize, FromRow)]
pub struct Cluster {
    pub id: Uuid,
    pub parish_id: Uuid,
    pub cluster_code: String,
    pub cluster_name: String,
    pub location_description: Option<String>,
    pub leader_name: Option<String>,
    pub is_active: Option<bool>,
    pub created_at: Option<DateTime<Utc>>,
    pub updated_at: Option<DateTime<Utc>>,
    pub deleted_at: Option<DateTime<Utc>>,
}

#[derive(Debug, Deserialize)]
pub struct CreateClusterRequest {
    pub parish_id: Uuid,
    pub cluster_code: String,
    pub cluster_name: String,
    pub location_description: Option<String>,
    pub leader_name: Option<String>,
}

#[derive(Debug, Deserialize)]
pub struct UpdateClusterRequest {
    pub cluster_name: Option<String>,
    pub location_description: Option<String>,
    pub leader_name: Option<String>,
    pub is_active: Option<bool>,
}

File: mobile/lib/models/cluster.dart
class Cluster {
  final String id;
  final String parishId;
  final String clusterCode;
  final String clusterName;
  final String? locationDescription;
  final String? leaderName;
  final bool? isActive;
  final String? createdAt;
  final String? updatedAt;
  final String? deletedAt;

  Cluster({
    required this.id,
    required this.parishId,
    required this.clusterCode,
    required this.clusterName,
    this.locationDescription,
    this.leaderName,
    this.isActive,
    this.createdAt,
    this.updatedAt,
    this.deletedAt,
  });

  factory Cluster.fromJson(Map<String, dynamic> json) {
    return Cluster(
      id: json['id'],
      parishId: json['parish_id'],
      clusterCode: json['cluster_code'],
      clusterName: json['cluster_name'],
      locationDescription: json['location_description'],
      leaderName: json['leader_name'],
      isActive: json['is_active'],
      createdAt: json['created_at'],
      updatedAt: json['updated_at'],
      deletedAt: json['deleted_at'],
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'parish_id': parishId,
      'cluster_code': clusterCode,
      'cluster_name': clusterName,
      'location_description': locationDescription,
      'leader_name': leaderName,
      'is_active': isActive,
      'created_at': createdAt,
      'updated_at': updatedAt,
      'deleted_at': deletedAt,
    };
  }
}

class CreateClusterRequest {
  final String parishId;
  final String clusterCode;
  final String clusterName;
  final String? locationDescription;
  final String? leaderName;

  CreateClusterRequest({
    required this.parishId,
    required this.clusterCode,
    required this.clusterName,
    this.locationDescription,
    this.leaderName,
  });

  Map<String, dynamic> toJson() {
    return {
      'parish_id': parishId,
      'cluster_code': clusterCode,
      'cluster_name': clusterName,
      'location_description': locationDescription,
      'leader_name': leaderName,
    };
  }
}
